ch_pipeline/analysis/flagging.py

```diff
@@ -1702,6 +1702,348 @@ def process(self, ss):
         return ss
 
 
+class MaskDecorrelatedCylinder(task.SingleTask):
+    """Identify and mask frequencies and times where a cylinder decorrelated.
+
+    Parameters
+    ----------
+    threshold: int
+        Mask frequencies and times where the median ratio of the
+        average-without-cylinder to average-with-cylinder visibility
+        amplitude is greater than this threshold.
+    max_frac_freq: float
+        Mask any frequency that was transmitted by an FPGA motherboard slot
+        with more than this fraction of frequencies masked by the above
+        threshold.  Only relevant if the frequency map is provided at setup.
+    """
+
+    threshold = config.Property(proptype=float, default=5.0)
+    max_frac_freq = config.Property(proptype=float, default=0.1)
+
+    def setup(self, freq_map=None):
+        """Determine the frequencies handled by each FPGA motherboard slot.
+
+        If a motherboard decorrelates, then all frequencies transmitted by
+        that slot will be affected.
+
+        Parameters
+        ----------
+        freq_map : FrequencyMap
+            The mapping between frequency bin and [crate, slot, link]
+            as a function of time.
+        """
+
+        self._set_slot_freqs(freq_map)
+
+    def process(self, data, inputmap):
+        """Create the mask.
+
+        Parameters
+        ----------
+        data : TimeStream
+            Visibilites before averaging over cylinders.
+        inputmap : list of :class:`CorrInput`
+            A list describing the inputs in data.
+
+        Returns
+        -------
+        out : RFIMask
+            Mask with True indicating that a cylinder decorrelated
+            at that frequency and time.
+        """
+
+        from draco.analysis.ringmapmaker import find_grid_indices
+
+        # Distribute over frequencies
+        data.redistribute("freq")
+
+        # Get polarisations of feeds
+        pol = tools.get_feed_polarisations(inputmap)
+
+        # Get positions of feeds
+        pos = tools.get_feed_positions(inputmap)
+
+        # Get cylinder each feed is on
+        cyl = np.array(
+            [chr(63 + inp.cyl) if tools.is_chime(inp) else "0" for inp in inputmap]
+        )
+        ucyl = np.unique(cyl[cyl != "0"])
+        ncyl = ucyl.size
```

Comment:
I think working with char's add a bunch of unnecessary complexity (especially as I don't think this task actually exposes any of the mapping to the user) and it makes some parts this PR harder to follow than they need to be. I would suggest just leave everything as ints if that can be made to work:
```suggestion
        # Get cylinder each feed is on (2 to 5 -> cyl_A to cyl_D)
        cyl = np.array([inp.cyl) if tools.is_chime(inp) else -1 for inp in inputmap])
        ucyl = np.unique(cyl[cyl < 0])
        ncyl = ucyl.size
```
I think there's one spot later down where a comparison needs to be changed. I'll attempt to add a suggestion there.

Reply:
Good point, there is no need to use chars in this task.